keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlUnhappyVetoRepository.kt

```diff
@@ -31,14 +32,30 @@ class SqlUnhappyVetoRepository(
   private val sqlRetry: SqlRetry
 ) : UnhappyVetoRepository(clock) {
 
-  override fun markUnhappyForWaitingTime(resourceId: String, application: String, wait: Duration) {
+  override fun markUnhappyForWaitingTime(resourceId: String, application: String, wait: Duration?) {
     sqlRetry.withRetry(WRITE) {
       jooq.insertInto(UNHAPPY_VETO)
         .set(UNHAPPY_VETO.RESOURCE_ID, resourceId)
         .set(UNHAPPY_VETO.APPLICATION, application)
-        .set(UNHAPPY_VETO.RECHECK_TIME, calculateExpirationTime(wait).toTimestamp())
+        .run {
+          calculateExpirationTime(wait)?.toTimestamp().let { expiryTime ->
+            if (expiryTime == null) {
```

Comment:
You're checking against nullability on the return of `calculateExpirationTime()`, so I assume any `Instant.toTimestamp()` would never be `null`?

Reply:
d'oh. Of course

Reply:
No, `toTimestamp` is only called if the result of `calculateExpirationTime` is not null. That means the overall  value of `calculateExpirationTime(wait)?.toTimestamp()` is `LocalDateTime?`.